Site Migration: Removing ending slash

gabrielcaires opened the pull request:
Closes  [#100865](https://github.com/Automattic/wp-calypso/issues/100865)

## Proposed Changes
* Remove the existing ending slash


## Testing Instructions
* Go to /hosted-site-migration
* Set any WP site using an ending slash (E.g https://snowshoe-of-stags.jurassic.ninja/)
*  Follow the assisted migration steps
*  When you arrive on the `Tell us about your WordPress site` page, check if the input is not showing the ending `/`
* Select the input and type `/`
*  Check if the code is not blocking you to type `/` if needed. 

![Uploading CleanShot 2025-03-12 at 18.07.15@2x.png…]()


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

gabrielcaires committed:
Removing ending slash

gabrielcaires commented:
The unit test suite is failing unrelated tests

sixhours reviewed (approved):
Works and doesn't mess up the URL when you try to add a slash :D

daledupreez commented:
@sixhours, might you be able to merge and deploy this PR today? If not, we should try to find someone else to get it shipped before the weekend.

daledupreez committed:
Only remove trailing slash when it is the only thing in the URL path

markbiek commented:
Just noting the unit test failures are real. I'm investigating.

markbiek committed:
Fix test failure caused by canParse being undefined in the test environment

This is caused by a node version-mismatch between what we're running
locally and what runs in the CI test environment.

markbiek reviewed (approved):
Noting that I had to make a small tweak to fix the test failure. The short version is that it seems to be related to the test environment running an older version of node.

I've retested the functionality and it's still working as expected!.

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/hooks/use-credentials-form.ts:
```diff
@@ -54,23 +54,27 @@ const isWPCOM = ( siteInfo?: UrlData ) => {
+			return url.slice( 0, -1 );
+		}
+	} catch {
+		if ( url.endsWith( '/' ) ) {
+			return url.slice( 0, -1 );
+		}
```
If it's not a valid URL, I think we should simply return it as-is, rather than trying to remove a trailing slash from some unexpected string.
